### PR TITLE
pow ffs sign and verify

### LINKED ISSUE #618: Include `pow ffs sign <challenge>` and `pow ffs verify <msg> <addr>` methods
Pow users are going to need the ability to sign arbitrary messages with their wallet identities. This will be a necessary step for getting rewarded in SR2. 

1. `pow ffs sign` would be great
2. `pow ffs verify` would complete the feature, though isn't required by SR2

### PR DESCRIPTION
**Pull request by jsign**
This PR includes two new `pow` CLI commands:
- `pow ffs sign`: which signs a message with each wallet address of an FFS instance.
- `pow ffs verify`: verifies signed messages from a wallet address. This wallet address can be any wallet address in the Filecoin network.

This behavior of signing with all possible wallets of the FFS instance is at the CLI level, the API has a more agnostic behavior to specify a single wallet address since that looks like the correct thing to do.

Closes #618

### CODE REVIEW

File: cmd/pow/cmd/ffs_sign.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		sb, err := hex.DecodeString(args[2])
+		checkErr(err)
+
+		s := spin.New(fmt.Sprintf("%s Verifying signature...", "%s"))
+		s.Start()
+		ok, err := fcClient.FFS.VerifyMessage(authCtx(ctx), args[0], mb, sb)
+		s.Stop()
+		checkErr(err)
+		if ok {
+			_, err := emoji.Println(":heavy_check_mark: The signature corresponds to the wallet address.")
```
Comment (asutula) [inline cmd/pow/cmd/ffs_sign.go:79]: Nice to see the emoji module used 😄

File: ffs/api/api_wallet.go [hunk 2]
```diff
@@ -60,6 +60,30 @@ func (i *API) NewAddr(ctx context.Context, name string, options ...NewAddressOpt
...
+	res, err := i.wm.Sign(ctx, addr, message)
+	if err != nil {
+		return nil, fmt.Errorf("signing message: %s", err)
+	}
+	return res, nil
+}
+
+// VerifyMessage verifies a message signature of a message from a managed address.
+func (i *API) VerifyMessage(ctx context.Context, addr string, message, signature []byte) (bool, error) {
+	if !i.isManagedAddress(addr) {
```
Comment (asutula) [inline ffs/api/api_wallet.go:77]: In the description for the PR that `VerifyMessage`:
> verifies signed messages from a wallet address. This wallet address can be any wallet address in the Filecoin network

Seems like you shouldn't enforce the provided address is managed. Would make sense this should be any address.

**Review by asutula — APPROVED** [review]
Looks good! One possible thing to fix.

File: cmd/pow/cmd/ffs_sign.go (hunk 1 above)
Comment (jsign) [reply cmd/pow/cmd/ffs_sign.go:79]: Yep, it's pretty cool. Less try CLI.

File: ffs/api/api_wallet.go (hunk 2 above)
Comment (jsign) [reply ffs/api/api_wallet.go:77]: Oh yeah... muscle memory. Will change.

**Pull request merged**